Source/WebKit/Platform/spi/visionos/WKFullScreenImmersiveSpaceDelegate.swift

```diff
@@ -0,0 +1,127 @@
+//Copyright (C) 2025 Apple Inc. All rights reserved.
+//
+//Redistribution and use in source and binary forms, with or without
+//modification, are permitted provided that the following conditions
+//are met:
+//1. Redistributions of source code must retain the above copyright
+//   notice, this list of conditions and the following disclaimer.
+//2. Redistribution in binary form must reproduce the above copyright
+//   notice, this list of conditions and the following disclaimer in the
+//   documentation and/or other materials provided with the distribution.
+//
+//THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+//AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+//BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+//CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+//SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+//INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+//CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+//ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+//THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+#if os(visionOS)
+
+import Combine
+@_weakLinked internal import SwiftUI
+import UIKit
+
+/// Manages surroundings effect state for fullscreen web content in visionOS.
+///
+/// This singleton coordinates the surroundings dimming effect between Objective-C/UIKit
+/// and SwiftUI layers, replacing the deprecated MRUIStage.preferredDarkness API.
+@MainActor
+@objc(WKFullScreenImmersiveSpaceManager)
+@available(visionOS 1.0, *)
+public class WKFullScreenImmersiveSpaceManager: NSObject, ObservableObject {
+    /// Shared singleton instance of the manager.
+    @objc
+    public static let shared = WKFullScreenImmersiveSpaceManager()
+
+    /// The current surroundings effect level (0=none, 1=semiDark, 2=dark, 3=ultraDark).
+    @Published
+    @objc
+    public dynamic var currentEffect: Int = 0
+
+    private override init() {
+        super.init()
+    }
+
+    var swiftUIEffect: SurroundingsEffect? {
```

Comment:
We're in SwiftUI code, and this property isn't exposed outside of SwiftUI code, so SwiftUI in the name doesn't add value. `fileprivate` because this variable is only in used in this file.

```suggestion
    fileprivate var surroundingsEffect: SurroundingsEffect? {
```